### DISCUSSION TITLE
Saving data after docker restart

### DISCUSSION THREAD

**Discussion by Walli-DO** [grafana/docker-otel-lgtm#127] [Q&A] [answered]
Hello, I used this solution and it turned out to be very convenient. But I didn’t find any description in the article of how you can point to “volumes” for grafana, loki, prometheuse and tempo. I checked and, as expected, all settings are reset after a reboot. I went inside and couldn’t find the data folder (/var/lib/grafana, ...).

Can you please tell me which folders are responsible for saving container data?

P.S: I also found something strange, you can easily log into Grafana without authorization and make all the settings, is this how it was intended? If so, is there any way to disable this?

**Comment by zeitlinger** [comment, marked as answer]
> P.S: I also found something strange, you can easily log into Grafana without authorization and make all the settings, is this how it was intended? If so, is there any way to disable this?

yes, this docker image is optimized for easy testing

you can pass grafana env vars to change that however - see https://github.com/grafana/grafana/blob/main/conf/sample.ini#L557

you can create a pr to set these env vars only if missing: https://github.com/grafana/docker-otel-lgtm/blob/b48b0954a3cfe7b830c3c2152c3c290a4441bf2e/docker/run-grafana.sh#L5-L6

this snippet also shows how env vars are translated, e.g. https://github.com/grafana/grafana/blob/main/conf/sample.ini#L632

> Can you please tell me which folders are responsible for saving container data?

see https://github.com/grafana/docker-otel-lgtm/blob/81745f67341e5106eea09e22e43b09dff4c3b888/run-lgtm.sh#L12-L15

**Reply by Walli-DO** [reply]
Got it thank you very much
